acos_client/v30/axapi_http.py

```diff
@@ -64,6 +64,16 @@ def request(self, method, api_url, params={}, headers=None,
         LOG.debug("axapi_http: %s url = %s", method, api_url)
         LOG.debug("axapi_http: params = %s", json.dumps(params, indent=4))
 
+        # Update params with **kwargs for currently unsupported configuration
+        # of objects
+        formatted_kwargs = dict([(k.replace('_', '-'), v) for k, v in kwargs.iteritems()])
+        param_keys = params.keys()
+        if params != {}:
```

Comment:
Certainly yours looks better.  But the latter lets you hand the customer the axapi reference and say, "extend it however you want." The yahoo driver gives them a way to pass axapi_args all the way from the neutron CLI.